### PR TITLE
Fix: coinjoin registered utxo selector

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Prerequisite for https://github.com/trezor/trezor-suite/pull/9021

TLDR; selector should exclude utxos from spending **only** if session is running **or** utxo is promised to successfully signed round (round id exists in transactionCandidates)

## Explanation

utxos stored in `CoinjoinPrison` have assigned `sentenceEnd` to prevent using them too often for example:
If utxo is successfully registered in Round and this round is not yet in critical phase you can still use "stop coinjoin" button.
This action **will not** unregister utxo from the round on coordinator, it will only stop sending `connection-confirmation` and coordinator will unregister it automatically (after ~45 seconds)
By clicking again on "start coinjoin" this utxo **should not** be registered again (until sentenceEnd) to the same or different round.
But it doesnt mean that this utxo is **not spendable** (excluded by selector) it only means that it shouldnt be used in coin registration.
Currently this utxo is **not spendable** until coinjoin package release it from the prison (until sentenceEnd) - which is a bug i discovered while working on `Coin control` feature.

Additionally i've renamed the selector to something more meaningful and also added some unit tests to increase coverage (it's very low already)

## Commits

1. fix in coinjoin package: release inmate from prison before `sentenceEnd` if assigned Round is no longler present in status 
    This means that utxo can be safely registered without `AliceAlreadyRegistered` error
2. fix in suite - described above
3. code hygiene - remove dead code. cj session is no longer stored in indexeddb therfore there is noo need to pause it

### CODE REVIEW

**Review by matejkriz — APPROVED** [review]
👏
